homeassistant/components/drop_connect/__init__.py

```diff
@@ -0,0 +1,66 @@
+"""The drop_connect integration."""
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+
+from .const import CONF_DATA_TOPIC, CONF_DEVICE_TYPE, DOMAIN
+from .coordinator import DROPDeviceDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up DROP from a config entry."""
+
+    # Make sure MQTT integration is enabled and the client is available.
+    if not await mqtt.async_wait_for_mqtt_client(hass):
+        _LOGGER.error("MQTT integration is not available")
+        return False
+
+    if TYPE_CHECKING:
+        assert config_entry.unique_id is not None
+    drop_data_coordinator = DROPDeviceDataUpdateCoordinator(
+        hass, config_entry.unique_id
+    )
+
+    @callback
+    def mqtt_callback(msg: ReceiveMessage) -> None:
+        """Pass MQTT payload to DROP API parser."""
+        if drop_data_coordinator.drop_api.parse_drop_message(
+            msg.topic, msg.payload, msg.qos, msg.retain
+        ):
+            drop_data_coordinator.async_set_updated_data(None)
+
+    config_entry.async_on_unload(
+        await mqtt.async_subscribe(
+            hass, config_entry.data[CONF_DATA_TOPIC], mqtt_callback, 0
+        )
```

Comment:
```suggestion
        await mqtt.async_subscribe(
            hass, config_entry.data[CONF_DATA_TOPIC], mqtt_callback
        )
```
QoS = 0 is the default, so no need to add it.

Reply:
Changed in ffe88053b5cd5eca3d48bd1ca820fcf6a327527d.

Reply:
Thnx